toolkit/data-sources/db-sync/src/db_model.rs

```diff
@@ -109,6 +109,17 @@ pub(crate) struct TokenTxOutput {
 	pub datum: Option<DbDatum>,
 }
 
+#[derive(Debug, Clone, sqlx::FromRow, PartialEq)]
+pub(crate) struct TokenDatumTxOutput {
+	pub origin_tx_hash: [u8; 32],
+	pub utxo_index: TxIndex,
+	pub tx_epoch_no: EpochNumber,
+	pub tx_block_no: BlockNumber,
+	pub tx_slot_no: SlotNumber,
+	pub tx_block_index: TxIndexInBlock,
+	pub datum: DbDatum,
+}
```

Comment:
I don't know if we need to get most of it from the database.
1. it will be a cost in data transfer
2. it will be a cost in writing tests for values we discard